plutus-tx-plugin/executables/profile/Main.hs

```diff
@@ -0,0 +1,101 @@
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications    #-}
+{-# OPTIONS_GHC -fplugin PlutusTx.Plugin #-}
+{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:defer-errors #-}
+{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:debug-context #-}
+{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:profile-all #-}
+
+-- | Executable for profiling. Add the program you want to profile here.
```

Comment:
I think we should be able to do this as a test, no? Can't we just have a test module with `profile-all` where we have golden tests with the profile output?

Reply:
Hmm, but it'll include timestamps. So we need a "test time" emitter that emits some kind of deterministic time. Maybe we could have one with an internal counter and it just increments it every time it emits something?

Reply:
Yes, with the timestamps the golden tests will fail. If we want to make sure the programs compile correctly we should do the golden tests separately? Like the existing infrastructure in `test`. 

Yes we can have one with a counter. However, in that case my script that will turn these files to flamegraphs won't have as much detail on how much time has passed. 

Or are you suggesting we have both? I was assuming we know these compile fine and we just need to do profiling.

Reply:
Well, I think we want some tests that the profiling machinery works properly, and for those the counter one would work. i.e. I'm suggesting we use the counter version _only_ in testing. Because there we mostly care that the sequencing is working. 

For actually playing with examples it maybe does make sense to have an executable, yeah.

Reply:
I see. Makes sense. I can do that.

Reply:
OK I think I've done it? Please see [this commit](https://github.com/input-output-hk/plutus/commit/917da1bc064f31b9f7c8ea01a7f77fd1aeff79d0). Are the golden files looking right to you?

Reply:
Nice! Having done all that, I've realised that.... well, the counter is irrelevant, right? It just goes up. We already get the ordering we need from the trace statements anyway! So we could have just used the non-timestamping trace emitter :facepalm: sorry for the wild goose chase!

Reply:
No worries! You've made it very easy to add another emitter! :) I'll switch to the log one and remove this emitter.